sotodlib/site_pipeline/make_cosamp_hk.py

```diff
@@ -0,0 +1,212 @@
+    else:
+        logger.info(f"Creating {man_db_filename} for the \
+                     archive index.")
+        scheme = core.metadata.ManifestScheme()
+        scheme.add_exact_match('obs:obs_id')
+        scheme.add_data_field('dataset')
+        man_db = core.metadata.ManifestDb(
+            man_db_filename,
+            scheme=scheme
+        )
+    
+    # query observations
+    if obs_id is not None:
+        tot_query = f"obs_id=='{obs_id}'"
+    else:
+        tot_query = "and "
+        if query_text is not None:
+            tot_query += f"{query_text} and "
+        if min_ctime is not None:
+            tot_query += f"timestamp>={min_ctime} and "
+        if max_ctime is not None:
+            tot_query += f"timestamp<={max_ctime} and "
+        tot_query = tot_query[4:-4]
+        if tot_query == "":
+            tot_query = "1"
+    
+    logger.info(f'tot_query: {tot_query}')
+    obs_list= ctx.obsdb.query(tot_query, query_tags)
+    
+    # output cosampled data for each obs_id
+    for obs in obs_list:
+        obs_id = obs['obs_id']
+        try:
+            meta = ctx.get_meta(obs_id)
+            aman = ctx.get_obs(obs_id, dets=[])
+            
+            
+            
+            _hkman = hk_utils.get_detcosamp_hkaman(aman, 
+                                      fields = hk2meta_config['fields'],
+                                      data_dir = hk2meta_config['input_dir'])
+            
+            dt_samp_hk = hk2meta_config['dt_samp_hk']
+            start = float(aman.timestamps[0] - 3*dt_samp_hk)
+            stop = float(aman.timestamps[-1] + 3*dt_samp_hk)
+            _data = hk_utils.sort_hkdata(start=start, stop=stop, fields=hk2meta_config['fields'],
+                                         data_dir=hk2meta_config['input_dir'], alias=None)
+            _hkman = hk_utils.make_hkaman(grouped_data=_data, alias_exists=False,
+                                          det_cosampled=True, det_aman=aman)
+            
+            hkman = core.AxisManager(meta.dets, aman.samps)
+            hkman.wrap('timestamps', aman.timestamps, [(0, 'samps')])
+
+            full_fields = hk2meta_config['fields']
+            aliases = hk2meta_config['aliases']
+            data_fields = [full_field.split('.')[1] for full_field in full_fields]
+            num_different_field = 0
+            for i, (full_field, data_field, alias) in enumerate(zip(full_fields, data_fields, aliases)):    
+                if i > 0:
+                    num_different_field += int(data_fields[i-1] != data_field)
+                i_in_same_field = i - num_different_field
+                hkman.wrap(alias, _hkman[data_field][data_field][i_in_same_field], [(0, 'samps')])
+            
+            h5_filename = f"{output_prefix}_{obs_id.split('_')[1][:4]}.h5"
+            output_filename = os.path.join(output_dir, h5_filename)
+            hkman.save(dest=output_filename, group=obs_id, overwrite=overwrite, compression='gzip')
```

Comment:
Do you need to gzip the data (i.e. do you think it's large enough for us to care)? I think that adds some extra io time

Reply:
Ah, good point. That should be fine. 
Actually I just copy and pasted from update-hwp-angle, but our case should be smaller than them.